### PR TITLE
 [17.0][OU-ADD] analytic: migration to 17.0

### PR DESCRIPTION
**Pull request by hbrunn**
https://github.com/OCA/OpenUpgrade/pull/4458 with review followup

### CODE REVIEW

**Comment by hbrunn** [comment]
/ocabot migration analytic

<dependency></dependency>

**Comment by hbrunn** [comment]
reminder: This needs to load the noupdate changes to be introduced in #4606

File: openupgrade_scripts/scripts/analytic/17.0.1.1/upgrade_analysis_work.txt [hunk 1]
```diff
@@ -0,0 +1,26 @@
+---Models in module 'analytic'---
+---Fields in module 'analytic'---
+analytic     / account.analytic.account / message_main_attachment_id (many2one): DEL relation: ir.attachment
+analytic     / account.analytic.account / root_plan_id (many2one)       : not a function anymore
+analytic     / account.analytic.account / root_plan_id (many2one)       : now related
+analytic     / account.analytic.plan    / _order                        : _order is now 'sequence asc, id' ('complete_name asc')
```
Comment (pedrobaeza) [inline openupgrade_scripts/scripts/analytic/17.0.1.1/upgrade_analysis_work.txt:10]: I'm checking now about this point, as we should preserve the order shown in previous version, but I'm not finding right now where the widget is taking the order. Investigating now...

**Comment by pedrobaeza** [comment]
Finish migration scripts with these changes:

- Set account.analytic.plan sequence according previous order (by id)
- Don't put default value for account.analytic.plan.applicability~company_id for preserving previous behavior.
- Better work file explanations.

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**